src/battle_main.c

```diff
@@ -1939,6 +1940,10 @@ u8 CreateNPCTrainerPartyFromTrainer(struct Pokemon *party, const struct Trainer
             u32 otIdType = OT_ID_RANDOM_NO_SHINY;
             u32 fixedOtId = 0;
             u32 ability = 0;
+            u16 species = partyData[i].species;
+
+            if (RANDOMIZER_AVAILABLE)
+                species = RandomizeTrainerMon(seed, i, monsCount, species);
```

Comment:
how does this work if there is a fully defined trainer mon with moves, nature, gender, etc?

Reply:
The trainer mon acts identically, except that its species is changed, if that makes sense. Hence you get stuff Brawly having a Shedinja that knows Rock Tomb.